diff --git a/giscanner/transformer.py b/giscanner/transformer.py
index bcabdedc..3ece97ce 100644
--- a/giscanner/transformer.py
+++ b/giscanner/transformer.py
@@ -321,15 +321,15 @@ raise ValueError."""
 
     def strip_identifier(self, ident):
         if self._identifier_filter_cmd:
-            proc = subprocess.Popen(self._identifier_filter_cmd,
-                                    stdin=subprocess.PIPE,
+            proc = subprocess.run(self._identifier_filter_cmd,
                                     stdout=subprocess.PIPE,
-                                    stderr=subprocess.PIPE)
-            proc_ident, err = proc.communicate(ident.encode())
+                                    stderr=subprocess.PIPE,
+                                    universal_newlines=True,
+                                    input=ident)
             if proc.returncode:
                 raise ValueError('filter: %r exited: %d with error: %s' %
-                                 (self._identifier_filter_cmd, proc.returncode, err))
-            ident = proc_ident.decode('ascii').strip()
+                                 (self._identifier_filter_cmd, proc.returncode, proc.stderr))
+            ident = proc.stdout.strip()
 
         hidden = ident.startswith('_')
         if hidden:

