diff --git a/girepository/meson.build b/girepository/meson.build
index 6a8c5b5..172be18 100644
--- a/girepository/meson.build
+++ b/girepository/meson.build
@@ -134,6 +134,7 @@ girepo_lib = shared_library('girepository-1.0',
                  gio_dep, girepo_internals_dep],
   version: '1.0.0',
   install: true,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
 )
 
 # Copy to builddir for use with giscanner/dumper.py when running uninstalled
diff --git a/giscanner/meson.build b/giscanner/meson.build
index 7b4b0d9..e0a5381 100644
--- a/giscanner/meson.build
+++ b/giscanner/meson.build
@@ -88,4 +88,5 @@ giscanner_pymod = python.extension_module('_giscanner', ['giscannermodule.c'],
                  python.dependency()],
   install: true,
   install_dir: giscannerdir,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
 )
diff --git a/tools/meson.build b/tools/meson.build
index 0bc33ba..635e9bf 100644
--- a/tools/meson.build
+++ b/tools/meson.build
@@ -44,6 +44,7 @@ gircompiler = executable('g-ir-compiler', 'compiler.c',
     girepo_dep,
   ],
   install: true,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
 )
 
 girgenerate = executable('g-ir-generate', 'generate.c',
@@ -52,9 +53,11 @@ girgenerate = executable('g-ir-generate', 'generate.c',
     girepo_dep,
   ],
   install: true,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
 )
 
 girinspect = executable('g-ir-inspect', 'g-ir-inspect.c',
   dependencies: girepo_dep,
   install: true,
+  install_rpath: join_paths(get_option('prefix'), get_option('libdir')),
 )
